linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/SJSGen.scala

```diff
@@ -151,29 +152,96 @@ private[emitter] final class SJSGen(
 
   def genSelect(receiver: Tree, field: irt.FieldIdent)(
       implicit pos: Position): Tree = {
-    DotSelect(receiver, Ident(genName(field.name))(field.pos))
+    DotSelect(receiver, genFieldIdent(field.name)(field.pos))
   }
 
-  def genSelect(receiver: Tree, field: irt.FieldIdent,
+  def genSelectWithOrigName(receiver: Tree, field: irt.FieldIdent,
```

Comment:
What is the rationale behind adding `WithOrigName` here? IIUC the overload already defines this (and has so in the past).

Was the intention to make the `originalName` parameter optional? (which I'm not sure is a good idea, feels easier to construct an `OriginalName` at the call site where this makes sense, which IIUC would only be the CoreJSLib).

Reply:
Ah, that's because I was *very* tempted to make the overload without `OriginalName` call the other one with `NoOriginalName`. However, that does *not* have the same effect. The former generates an ident that *never* has a name binding in the source map, while the latter may still generate a name binding if the produced ident does not match the source ident.

So I made it clear with the `WithOrigName` suffix that these methods do something *different*.

Reply:
I think this makes sense: It is too easy to think from just looking at `SJSGen` that we just put original names on all idents where we have them.

Maybe it makes sense to put a comment at the top explaining the `WithOrigName` suffix? Alternatively, we could call it `ForDef`, making its intent clearer (IIUC we even call `WithOrigName` if `OriginalName == NoOriginalName`, which seems like an oxymoron).

Also: Is this a refactor that is worth putting in a separate commit?

Reply:
+1 for `ForDef`, and I extracted the refactoring in a separate commit.